### PR TITLE
220 refactoring codi backend

### PR DESCRIPTION
**Pull request by aolle99**

### CODE REVIEW

File: api/bookings/serializers.py [hunk 1]
```diff
@@ -1,48 +1,67 @@
...
-            return None
+    def validate(self, attrs):
+        start_date = attrs['start_date']
+        end_date = attrs['end_date']
+        publication = Publication.objects.get(id=attrs['publication'].id)
+        if start_date > end_date:
+            raise serializers.ValidationError("Start date must be before end date")
+        if start_date < datetime.now():
+            raise serializers.ValidationError("Start date must be in the future")
+        if start_date < datetime.now():
```
Comment (crismigo) [inline api/bookings/serializers.py:47]: start_date per end_date

File: api/ratings/services.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return ratings
+
+def create_post_rating(rating):
+    booking = Bookings.objects.get(id=rating["booking"])
+    if booking.user.id != rating["user"]:
+        raise Exception("User is not client of the booking")
+    if booking.end_date > datetime.now():
+        raise Exception("Booking is not finished")
+    serializer = PostRatingSerializer(data=rating)
+    print(serializer)
```
Comment (crismigo) [inline api/ratings/services.py:27]: treure

File: api/bookings/serializers.py (hunk 1 above)
Comment (crismigo) [inline api/bookings/serializers.py:47]: start_date per end_date

**Review by crismigo — CHANGES_REQUESTED** [review]

File: api/ratings/services.py (hunk 2 above)
Comment (aolle99) [reply api/ratings/services.py:27]:
```suggestion
```

File: api/bookings/serializers.py (hunk 1 above)
Comment (aolle99) [reply api/bookings/serializers.py:47]:
```suggestion
        if end_date< datetime.now():
```

**Review by crismigo — APPROVED** [review]

**Pull request merged**